client/src/api/SSOTretrieval.js

```diff
@@ -7,24 +7,48 @@
  * @description Fetch all those SSOT names and Ids, which are available for the current user
  * @param {String} userId - String including the user Id
  */
- const fetchSSOTsForUser = async (userId) => {
-    const requestString = `/ssot/getAvailableBotsForUser/${ userId }`;
-    const response = await fetch(requestString);
-    return response;
-  };
+const fetchSSOTsForUser = async (userId) => {
+  const requestString = `/ssot/getAvailableBotsForUser/${userId}`;
+  const response = await fetch(requestString);
+  return response;
+};
+
+/**
+ * @description This function renames the robot in SSOT
+ * @param {String} SSOTid - String including the SSotid
+ * @param {String} newName - String with the new RobotName
+ */
+const changeSSOTName = async (SSOTid, newName) => {
+  const adjustedName = newName.replace(/\s/g, '+');
+  const requestString = `/ssot/renameBot?id=${SSOTid}&newName=${adjustedName}`;
+  const response = await fetch(requestString);
+  return response;
+};
+
+/**
+ * @description Fetches all the Metadata for a single Robot
+ * @param {String} robotId - String including the robotId
+ */
+const retrieveMetadataForBot = async (robotId) => {
+  const requestString = `/ssot/retrieveMetadataForBot/${robotId}`;
+  const response = await fetch(requestString);
+  return response;
+};
 
-  /**
-   * @description Fetch all those SSOT names and Ids, which are available for the current user
-   * @param {String} userId - String including the user Id
-   */
-   const changeSSOTName = async (SSOTid, newName) => {
-      const adjustedName = newName.replace(/\s/g, '+');
-      const requestString = `/ssot/renameBot?id=${ SSOTid }&newName=${ adjustedName }`;
-      const response = await fetch(requestString);
-      return response;
-    };
+/**
+ * @description Create a new robot with the specified name for the specified user 
+ * @param {String} newName - String including the user Id
+ */
+const createNewBot = async (userid, newName) => {
+  const adjustedName = newName.replace(/\s/g, '+');
+  const requestString = `/ssot/createNewBot?userid=${userid}&botName=${adjustedName}`;
+  const response = await fetch(requestString);
+  return response;
+};
```

Comment:
Why is the use of newName necessary here? Shouldn't the name be set with the call that we use to change the name?

Reply:
In principle you are right, however we have directly prepared the backend and also this frontend function to receive a parameter from us. This makes it easier to implement the background functionality behind the CreateNewRobotPopup. (#169)